### DISCUSSION TITLE
(ffmpeg-full) use full-shared

### DISCUSSION THREAD

**Discussion by nosovk** [chocolatey-community/chocolatey-packages#2292] [Ideas]
### Before starting, ensure that you have done the following.

- [X] I have verified that there is no other issue or discussion covering the feature or enhancement
- [X] I have verified there is no open pull request for the feature or enhancement
- [X] I have verified the package to improve is located in this repository

### What do you want to add?

I suggest adding ffmpeg-shared DLL's to ffmpeg-full package.

### What problem will the feature/enhancement solve?

Currently, when we install `ffmpeg-full` it uses all in one exe file ffmpeg version. It's not bad, it's just not obvious that DLLs from ffmpeg are not installed.
For example, you need [ffmpeg to work with audacity](https://support.audacityteam.org/basics/installing-ffmpeg). And you can't use any of choco version with audacity from choco, becasue both "essentials" and "full" versions installed without DLL's.

https://ffmpeg.org/platform.html
https://www.reddit.com/r/ffmpeg/comments/r2owve/what_are_the_differences_between_fullbuild_and/

### If the feature or enhancement is approved:

- [ ] I am willing to create a pull request to implement the feature or enhancement

**Comment by pauby** [comment]
[From this page](https://lame.buanzo.org/ffmpeg.php) it looks like Audacity requires a special version of ffmpeg.

**Reply by nosovk** [reply]
https://www.gyan.dev/ffmpeg/builds/
<img width="464" alt="image" src="https://github.com/chocolatey-community/chocolatey-packages/assets/1398808/bf612e0c-aa8e-46cb-a618-3cfa6649fcea">
and in a bottom of page:
<img width="563" alt="image" src="https://github.com/chocolatey-community/chocolatey-packages/assets/1398808/35750191-0e72-4c84-944d-a9745fe28079">
https://www.gyan.dev/ffmpeg/builds/packages/ffmpeg-5.1.2-full_build-shared.7z

I mean it's standart build, just not bundled in one file.

**Reply by pauby** [reply]
Is that an official download location? The one I linked to came from the official page.

**Reply by nosovk** [reply]
yes, this is official location
<img width="476" alt="image" src="https://github.com/chocolatey-community/chocolatey-packages/assets/1398808/de969d42-5041-4354-b5dd-7d6badd12b82">

**Reply by nosovk** [reply]
Or if you are talking about Audacity - there is no special version for Audacity. It works with official ffmpeg build

**Reply by pauby** [reply]
As the package currently uses an expected installer, if you want to switch to a different one I'd suggest you create a different package.

@VuiMuich you are listed in the [CODEOWNERS file](https://github.com/chocolatey-community/chocolatey-packages/blob/master/.github/CODEOWNERS#L31C23-L31C32) for `ffmpeg-full`. Can you share any thoughts on this?

**Reply by VuiMuich** [reply]
Tbh. currently I am a bit out of the loop in regards of chocolate and won't be able to meaningfully look into this matter in the next two weeks as I won't have access to a windows machine. 

The reason, why I packaged the `ffmpeg-full` was mainly for the encoding ability with the `HAP` codec. I don't really know, what the requirements of `audacity` actually is.

**Reply by nosovk** [reply]
audacity was just an example, in a real life the problem that we offer bundled setup, but not libraries. As a developer I need to use ffmpeg libraries directly time to time, and it would be nice to have some option to install both libraries and exe from choco.

**Reply by pauby** [reply]
My suggestion would be to create the packages for that. Most people just want to install the software itself and that's what this community repository currently caters for.

**Reply by nosovk** [reply]
like `ffmpeg-full-shared`?

**Reply by pauby** [reply]
Or `ffmpeg-full-library`. Just as an FYI, this would likely have to be hosted yourself as I don't see it being accepted into the Chocolatey Community Chocolatey Packages repository.

**Reply by nosovk** [reply]
no problem, I've installed it manually because I needed those libs to be global available, but users can make installation via scoop if user needs local installation. You may see that it's in a list of official installation methods for release-full-shared option.

**Comment by pauby** [comment]
Thanks @VuiMuich. Given the situation I would still suggest a separate package is created for use with Audacity as what is currently packaged is expected by the end user. If somebody else has any thoughts to share on this, please do.

**Comment by TheCakeIsNaOH** [comment]
There is already a `ffmpeg-audacity` package here: https://community.chocolatey.org/packages/audacity-ffmpeg
Although since this ffmpeg-full-shared build also works with audacity, and presumably has more build features enabled, then there seems to be no reason it can't be put into a new package.

**Reply by pauby** [reply]
That would mean we end up with an `ffmpeg-full`, `audacity-ffmepg` and `ffmpeg-somethingelse`?
